src/stream/src/executor/source.rs

```diff
@@ -284,19 +318,44 @@ impl ExecutorV1 for SourceExecutor {
                     .unwrap();
 
                 // todo: use epoch from msg to restore state from state store
-
                 assert!(matches!(msg, Message::Barrier(_)));
+                let epoch_prev = match &msg {
+                    Message::Barrier(barrier) => barrier.epoch.prev,
+                    _ => {
+                        return Err(RwError::from(InternalError(
+                            "expected barrier when first calling next, got data chunk".to_string(),
+                        )))
+                    }
+                };
+
+                let connector_type = match self.source_desc.source.as_ref() {
+                    SourceImpl::Connector(source) => source.config.get_connector_type()?,
+                    _ => "".to_string(),
```

Comment:
```suggestion
                    _ => todo!(),
```

Reply:
actually, it's intentional, table source here doesn't need to load from the state store.

Reply:
then use `let source = try_match_expand!(self.source_desc.source.as_ref(), SourceImpl::Connector).unwrap()`;